### ISSUE TITLE
regarding patch wise training and convolutional training

### ISSUE THREAD

**Issue by wenouyang** [shekkizh/FCN.tensorflow#10]
Hi Sarath,

Thanks for sharing the code.  I have a question regarding generating the training data set.

In the FCN paper, the authors discuss the patch wise training and fully convolutional training. What is the difference between these two?

Please refer to section 4.4 attached in the following.

It seems to me that the training mechanism is as follows, Assume the original image is M*M, then iterate the M*M pixels to extract N*N patch (where N<M). The iteration stride can some number like N/3 to generate overlapping patches. Moreover, assume each single image corresponds to 20 patches, then we can put these 20 patches or 60 patches(if we want to have 3 images) into a single mini-batch for training. Is this understanding right? It seems to me that this so-called fully convolutional training is the same as patch-wise training.

![capture](https://cloud.githubusercontent.com/assets/5430158/23634336/5075bffa-028f-11e7-8656-e308a9e0d28f.JPG)

**Comment by shekkizh** [comment]
Hi,
The distinction is in the mini batch loss used for training.
In your definition while constructing mini batch you suggest using all the patches in the same image together - this is what happens in FCNs. Whereas previously to introduce higher variance in mini batches people resorted to using patches from different images together in a mini batch i.e a mini batch need not contain all the patches corresponding to a single image. 

The authors argue that such sampling schemes doesn't really make a difference in training. Hope that makes sense.
